sunpy/map/mapbase.py

```diff
@@ -2154,19 +2157,32 @@ def draw_quadrangle(self, bottom_left, *, width: u.deg = None, height: u.deg = N
         """
         axes = self._check_axes(axes, allow_non_wcsaxes=False)
 
-        bottom_left, top_right = get_rectangle_coordinates(
-            bottom_left, top_right=top_right, width=width, height=height)
+        if isinstance(bottom_left, astropy.units.quantity.Quantity):
+            pixel_corners = u.Quantity(self._parse_submap_input(bottom_left, top_right, width, height)).T
+            bottom = np.min(pixel_corners[1])
+            top = np.max(pixel_corners[1])
+            left = np.min(pixel_corners[0])
+            right = np.max(pixel_corners[0])
+            width = (right-left)
+            height = (top-bottom)
+            anchor = (bottom, left)*u.pix
+            kwargs.update({"vertex_unit": u.pix})
```

Comment:
Is there no way to restructure this?

Reply:
Drawn parallels from `submap`, I suppose we could ignore the np.max and np.min and rewrite this but it depends on how the user passes bottom_left, width, height etc

Reply:
```suggestion
            anchor, _, top_right, _ = self._parse_submap_quantity_input(bottom_left, top_right, width, height)
            width, height = top_right - anchor
            kwargs.update({"vertex_unit": u.pix})
```
Unlike `submap()`, we are fine with negative `width` or negative `height` values.

Reply:
Yeah that makes sense, I shall make that change